ass/document.py

```diff
@@ -578,3 +432,247 @@ class Command(_Event):
     """ A command event. Not widely supported.
     """
     TYPE = "Command"
+
+class CaseInsensitiveOrderedDict(collections.abc.MutableMapping):
```

Comment:
Doesn't implement `__contains__`, but that's polyfilled.

Reply:
It's not listed as an abstract method in the `collections.abc` documentation, so I don't think it's needed.

Reply:
It's an abstract method of `Collection`, which `Mapping` (which `MutableMapping` inherits) polyfills. Probably using `__iter__`.